### ISSUE TITLE
Regression: inputHasAlpha: false no longer working

### ISSUE THREAD

**Issue by josephg** [lukeapage/pngjs#93]
PNGJS 3.3 no longer respects inputHasAlpha: false and is assuming my input data has a stride of 4 instead of 3.

Given this code:

```javascript
const PNG = require('pngjs').PNG
const W = 20, H = 20
const imgBuffer = new Buffer(W * H * 3)

for (let y = 0; y < H; y++) {
  for (let x = 0; x < W; x++) {
    const px = y * W + x
    
    imgBuffer[px*3] = 255
    imgBuffer[px*3+1] = 0
    imgBuffer[px*3+2] = 0
  }
}

const img = new PNG({
  width: W, height: H,
  colorType: 2, // color but no alpha
  bitDepth: 8,
  inputHasAlpha: false,
})

img.data = imgBuffer
img.pack().pipe(process.stdout)
```

With pngjs 3.2 I get the correct output (red):
![32](https://user-images.githubusercontent.com/47413/32782281-044cff24-c99c-11e7-9993-9680219cb065.png)

With pngjs 3.3 I get this:
![33](https://user-images.githubusercontent.com/47413/32782282-047ce176-c99c-11e7-8e6b-8cfa0c4c724e.png)

**Comment by josephg** [comment]
This is fixed with `inputColorType: 2`, but an API change like this should only happen when the major version changes. (I'm pegged to `pngjs^3` and my code broke!)

**Comment by lukeapage** [comment]
Sorry about that, it was unintentional, I expect that was a result of @jnuevo changes. I will merge and publish a fix if someone provides a pr, however I do not actively use this module any more so I won’t be fixing it myself.

**Comment by jnuevo** [comment]
@josephg @lukeapage I'll have a look at this one and submit a pr with a fix

**Comment by jnuevo** [comment]
The issue is that inputColorType takes a value of 6 by default (RGBA), and inputHasAlpha is ignored in 3.3 in some cases like this one. (`inputHasAlpha` is kind of redundant when you have an `inputColorType` option that can signal that and more.).

I've added a fix in PR #94, please let me know if you encounter other issues

**Comment by josephg** [comment]
🎉 thanks!
